src/buffer.js

```diff
@@ -194,14 +198,29 @@
             return;
         }
 
+        var requested_start = offset;
+        var requested_length = len;
+        if(this.fixed_chunk_size)
+        {
+            requested_start = offset - (offset % this.fixed_chunk_size);
+            requested_length = (Math.floor((offset + len - 1) / this.fixed_chunk_size) + 1 - Math.floor(requested_start / this.fixed_chunk_size)) * this.fixed_chunk_size;
```

Comment:
I think this would be more readable (and in line with `AsyncXHRPartfileBuffer.prototype.get`):

```js
requested_length = Math.ceil((offset - requested_start + len) / this.fixed_chunk_size) * this.fixed_chunk_size;
```

Reply:
Done!